### PR TITLE
Updates for GEFS v12.3

### PR DESCRIPTION
**Pull request by lipan-NOAA**
This is the code changes to address issue https://github.com/NOAA-EMC/GSDCHEM/issues/4 , which includes

- The fengsha bug fix corrects the calculation of the (kok, 2011) volume distribution of emitted dust increasing the mass of the fine mode dust
- modified the large wet removal process of aerosols
- adjusted SO2 anthropogenic emission factor
- adjusted OC and BC emissions processing in GBBEPx

### CODE REVIEW

**Comment by lipan-NOAA** [comment]
@WalterKolczynski-NOAA @XianwuXue-NOAA @rmontuoro @bbakernoaa @zhanglikate 
This is the GSDCHEM repository PR for GEFS-Aerosols v12.3. 
@WalterKolczynski-NOAA Are you the owner of this repository ?

File: src/chem/dep/dep_wet_ls_mod.F90 [hunk 1]
```diff
@@ -51,6 +51,18 @@ subroutine dep_wet_ls_init(config, rc)
         select case (config % chem_opt)
           case (CHEM_OPT_GOCART)
             alpha = 1.0
+            alpha(p_sulf) = 0.5
+            alpha(p_bc1) = 0.3  !lpan
```
Comment (XianwuXue-NOAA) [inline src/chem/dep/dep_wet_ls_mod.F90]: Why do you add !lpan, if needed, I think you may add more description.

File: src/chem/dep/dep_wet_ls_mod.F90 [hunk 2]
```diff
@@ -153,14 +164,18 @@ subroutine wetdep_ls(dt,var,rain,moist,rho,var_rmv,           &
 !
 ! proportionality constant
 !
-    frc(:,:)=0.1
+    !frc(:,:)=0.1
```
Comment (XianwuXue-NOAA) [inline src/chem/dep/dep_wet_ls_mod.F90]: If not needed, you can delete the unused lines.

File: src/chem/dep/dep_wet_ls_mod.F90 [hunk 3]
```diff
@@ -169,7 +184,19 @@ subroutine wetdep_ls(dt,var,rain,moist,rho,var_rmv,           &
         do k=1,kte
            dvar=max(0.,(moist(i,k,j,p_qc)+moist(i,k,j,p_qi)))
            var_sum_clw(i,j)=var_sum_clw(i,j)+dvar
+           var_sum(i,j)=var_sum(i,j)+var(i,k,j,nv)*rho(i,k,j) !lzhang
```
Comment (XianwuXue-NOAA) [inline src/chem/dep/dep_wet_ls_mod.F90]: Delete the !lzhang or add more detailed description.

File: src/chem/dep/dep_wet_ls_mod.F90 [hunk 4]
```diff
@@ -178,22 +205,26 @@ subroutine wetdep_ls(dt,var,rain,moist,rho,var_rmv,           &
...
            dvar=var(i,k,j,nv)-1.e-16
+           var_rmvl(i,k,j)=dvar !lzhang
            var(i,k,j,nv)=var(i,k,j,nv)-dvar
         else
            var(i,k,j,nv)=var(i,k,j,nv)-dvar
         endif
-        var_rmv(i,j,nv)=var_rmv(i,j,nv)+var_rmvl(i,k,j)
+        !var_rmv(i,j,nv)=var_rmv(i,j,nv)+var_rmvl(i,k,j)
+        !!convert wetdeposition into ug/m2/s  
+        var_rmv(i,j,nv)=var_rmv(i,j,nv)+(var_rmvl(i,k,j)*rho(i,k,j)*dz8w(i,k,j)/dt) !lzhang
```
Comment (XianwuXue-NOAA) [inline src/chem/dep/dep_wet_ls_mod.F90]: Delete unused lines

File: src/chem/dep/dep_wet_ls_mod.F90 [hunk 5]
```diff
@@ -303,7 +334,8 @@ subroutine WetRemovalGOCART ( i1, i2, j1, j2, k1, k2, n1, n2, cdt, &
 !  Accumulate the 3-dimensional arrays of rhoa and pdog
    do j = j1, j2
     do i = i1, i2
-      pdog(i,k1:k2,j) = (ple(i,k1+1:k2+1,j)-ple(i,k1:k2,j)) / grav
+      !pdog(i,k1:k2,j) = (ple(i,k1+1:k2+1,j)-ple(i,k1:k2,j)) / grav
```
Comment (XianwuXue-NOAA) [inline src/chem/dep/dep_wet_ls_mod.F90]: Delete unused lines

File: src/chem/dep/dep_wet_ls_mod.F90 [hunk 6]
```diff
@@ -326,25 +358,30 @@ subroutine WetRemovalGOCART ( i1, i2, j1, j2, k1, k2, n1, n2, cdt, &
 
 !    Find the highest model layer experiencing rainout.  Assumes no
 !    scavenging if T < 258 K
-     LH = 0
-     do k = k1, k2
+     !LH = 0
+     LH = k2+1 !lzhang
+     !do k = k1, k2
```
Comment (XianwuXue-NOAA) [inline src/chem/dep/dep_wet_ls_mod.F90]: Same comments as before

**Review by XianwuXue-NOAA — COMMENTED** [review]
I think you may delete the unused statements or add more detailed description of the line

**Comment by XianwuXue-NOAA** [comment]
@WalterKolczynski-NOAA Could you also review this PR for GEFSv12.3.0 aerosol fix?

**Comment by XianwuXue-NOAA** [comment]
@WalterKolczynski-NOAA could you please also add @zhanglikate and @bbakernoaa as reviewer because we don't have permission.

**Review by WalterKolczynski-NOAA — CHANGES_REQUESTED** [review]
Much the same as what Xianwu already said: remove the unused lines and the author identifiers.

File: src/chem/dep/dep_wet_ls_mod.F90 (hunk 2 above)
Comment (zhanglikate) [reply src/chem/dep/dep_wet_ls_mod.F90]: agree

File: src/chem/dep/dep_wet_ls_mod.F90 (hunk 3 above)
Comment (zhanglikate) [reply src/chem/dep/dep_wet_ls_mod.F90]: agree

File: src/chem/dep/dep_wet_ls_mod.F90 (hunk 4 above)
Comment (zhanglikate) [reply src/chem/dep/dep_wet_ls_mod.F90]: agree

**Review by zhanglikate — COMMENTED** [review]
Please also update dust_afwa_mod.F90.

File: src/chem/dep/dep_wet_ls_mod.F90 (hunk 1 above)
Comment (lipan-NOAA) [reply src/chem/dep/dep_wet_ls_mod.F90]: removed it

File: src/chem/dep/dep_wet_ls_mod.F90 (hunk 2 above)
Comment (lipan-NOAA) [reply src/chem/dep/dep_wet_ls_mod.F90]: removed it

**Comment by lipan-NOAA** [comment]
@XianwuXue-NOAA @WalterKolczynski-NOAA 
a new commit was pushed

File: src/chem/dep/dep_wet_ls_mod.F90 [hunk 7]
```diff
@@ -169,7 +179,19 @@ subroutine wetdep_ls(dt,var,rain,moist,rho,var_rmv,           &
...
            dvar=max(0.,(moist(i,k,j,p_qc)+moist(i,k,j,p_qi)))
            var_sum_clw(i,j)=var_sum_clw(i,j)+dvar
+           var_sum(i,j)=var_sum(i,j)+var(i,k,j,nv)*rho(i,k,j)
         enddo
+           if(var_sum(i,j).gt.1.e-10 .and. var_sum_clw(i,j).gt.1.e-10 ) then
+   !        assuming that frc is onstant, it is my conversion factor 
+!       (just like in convec. parameterization
+              frc(i,j)=rain_clw(i,j)/var_sum_clw(i,j)
+!    write(0,*)'frc ', frc(i,j),var_sum_clw(i,j),var_sum(i,j)
+              if (lat(i,j)<=-55.) then
```
Comment (WalterKolczynski-NOAA) [inline src/chem/dep/dep_wet_ls_mod.F90:189]: This block isn't properly indented

File: src/chem/dep/dep_wet_ls_mod.F90 [hunk 8]
```diff
@@ -326,25 +351,26 @@ subroutine WetRemovalGOCART ( i1, i2, j1, j2, k1, k2, n1, n2, cdt, &
 
 !    Find the highest model layer experiencing rainout.  Assumes no
 !    scavenging if T < 258 K
-     LH = 0
-     do k = k1, k2
+     !LH = 0
```
Comment (WalterKolczynski-NOAA) [inline src/chem/dep/dep_wet_ls_mod.F90]: Remove dead code

File: src/chem/dep/dep_wet_ls_mod.F90 [hunk 9]
```diff
@@ -326,25 +351,26 @@ subroutine WetRemovalGOCART ( i1, i2, j1, j2, k1, k2, n1, n2, cdt, &
...
+     LH = k2+1
+     do k = k2, k1,-1
```
Comment (WalterKolczynski-NOAA) [inline src/chem/dep/dep_wet_ls_mod.F90]: Looks like you should have a space after the last comma?

File: src/chem/dust/dust_fengsha_mod.F90 [hunk 10]
```diff
@@ -442,7 +442,7 @@ SUBROUTINE source_dust(imx, jmx, lmx, nmx, smx, dt1, tc, ustar, massfrac, &
             EXP(-(2.0*reff_dust(n)/lambda)**3.0)*dlndp(n)
        dvol_tot=dvol_tot+dvol(n)
        ! Convert mass flux to volume flux
-       emit_vol=emit/den_dust(n) ! (m s^-1)
+       ! emit_vol=emit/den_dust(n) ! (m s^-1)
```
Comment (WalterKolczynski-NOAA) [inline src/chem/dust/dust_fengsha_mod.F90]: Remove dead code

File: src/chem/dust/dust_fengsha_mod.F90 [hunk 11]
```diff
@@ -455,13 +455,14 @@ SUBROUTINE source_dust(imx, jmx, lmx, nmx, smx, dt1, tc, ustar, massfrac, &
...
              ! Calculate total mass emitted
-             dsrc = emit_vol*den_dust(n)*distr_dust(n)*dxy(j)*dt1  ! (kg)
+             dsrc = emit*distr_dust(n)*dxy(j)*dt1  ! (kg)
              IF (dsrc < 0.0) dsrc = 0.0
 
              ! Update dust mixing ratio at first model level.
              tc(i,j,1,n) = tc(i,j,1,n) + dsrc / airmas(i,j,1) ! (kg/kg)
              !   bems(i,j,n) = dsrc  ! diagnostic
-             bems(i,j,n) = 1000.*dsrc/(dxy(j)*dt1) ! diagnostic (g/m2/s)
+             !bems(i,j,n) = 1000.*dsrc/(dxy(j)*dt1) ! diagnostic (g/m2/s)
```
Comment (WalterKolczynski-NOAA) [inline src/chem/dust/dust_fengsha_mod.F90]: Remove dead code

**Review by WalterKolczynski-NOAA — CHANGES_REQUESTED** [review]
I'm not qualified to comment on the science changes here, but it looks like there are some style inconsistencies that should be fixed.

**Review by XianwuXue-NOAA — COMMENTED** [review]
Agree with Walter's comments

**Comment by lipan-NOAA** [comment]
@WalterKolczynski-NOAA @XianwuXue-NOAA  the code was modified, please review it

**Comment by lipan-NOAA** [comment]
@bbakernoaa Would you review it ?

**Review by bbakernoaa — APPROVED** [review]

**Comment by lipan-NOAA** [comment]
@WalterKolczynski-NOAA would you merge it?

**Review by WalterKolczynski-NOAA — APPROVED** [review]

**Comment by WalterKolczynski-NOAA** [comment]
@lipan-NOAA Can you improve the commit message on this PR first? Doesn't really say what was done.

**Comment by lipan-NOAA** [comment]
@WalterKolczynski-NOAA which one do you want me to change ?

**Comment by WalterKolczynski-NOAA** [comment]
> @WalterKolczynski-NOAA which one do you want me to change ?

The title and the description in the first comment.

Also, in the future, this should be four issues and four separate PRs.

**Comment by lipan-NOAA** [comment]
@WalterKolczynski-NOAA I did it Thanks

**Comment by WalterKolczynski-NOAA** [comment]
I see no change in the title or first comment of the PR.

**Comment by lipan-NOAA** [comment]
@WalterKolczynski-NOAA 
https://github.com/NOAA-EMC/GSDCHEM/pull/6/commits/a5929505a450a9a7c8b3cc61ab5d2914e2af31e4
"The fengsha bug fix corrects the calculation of the (kok, 2011) volume
    distribution of emitted dust increasing the mass of the fine mode dust"

**Comment by WalterKolczynski-NOAA** [comment]
Needs to be here:
image

**Comment by lipan-NOAA** [comment]
@WalterKolczynski-NOAA sorry for misunderstanding you. I modified it.

**Comment by WalterKolczynski-NOAA** [comment]
Will merge this today. I assume we also need a tag?

**Comment by lipan-NOAA** [comment]
@WalterKolczynski-NOAA Thanks

**Comment by XianwuXue-NOAA** [comment]
> Will merge this today. I assume we also need a tag?

In the previous package, we don't create a tag, however, I think creating a tag is more clear. How about use gefs_v12.2.0 as tag name?

**Comment by WalterKolczynski-NOAA** [comment]
> > Will merge this today. I assume we also need a tag?
> 
> In the previous package, we don't create a tag, however, I think creating a tag is more clear. How about use gefs_v12.2.0 as tag name?


Any important version, especially ones going to ops, should have a tag.

Just to be sure, this is for GEFS v12.2 and not GEFS v12.3? Everything else points towards 12.3.

**Comment by XianwuXue-NOAA** [comment]
> > > Will merge this today. I assume we also need a tag?
> > 
> > 
> > In the previous package, we don't create a tag, however, I think creating a tag is more clear. How about use gefs_v12.2.0 as tag name?
> 
> Any important version, especially ones going to ops, should have a tag.
> 
> Just to be sure, this is for GEFS v12.2 and not GEFS v12.3? Everything else points towards 12.3.

Typo. Yes, it should be 12.3

**Pull request merged**